brave/browser-laptop
# Add extensionsTab to preferences page

Add extensionsTab to preferences page

Auditors: @bsclifton, @bbondy

Closes #6530

<img width="1270" alt="screen shot 2017-04-04 at 4 10 08 pm" src="https://cloud.githubusercontent.com/assets/4672033/24674271/3ec808d4-1951-11e7-8480-21987152f7a9.png">

## Test plan:

1. Automated test must pass:

_extension state_
```
npm run test -- --grep="extensionUninstalled"
```

_extensionsReducer_
```
npm run test -- --grep="extensionsReducer"
```

_extensionsTab_
```
npm run test -- --grep="ExtensionsTab component"
```

## QA Steps:

1. Go to extensions tab
2. Enabling an extension should enable it
3. Disabling an extension should disable it
4. Removing an extension should remove its folder (you can see its path logged on console)
5. Removed extension should not appear in UI, and should be disabled after removal
6. Enabling a removed extension should install it again
7. Enabling another password manager should disable the first
8. Disabling all password managers should set "built-in" password manager to active
9. You can't remove built-in (webtorrent and sync) extensions -- **update** per @bradleyrichter request Sync is no longer included
